packages/kotti-ui/source/kotti-field-inline-edit/KtFieldInlineEdit.vue

```diff
@@ -0,0 +1,356 @@
+	name: 'KtFieldInlineEdit',
+	components: {
+		ConfirmButton,
+		EditIcon,
+	},
+	props: makeProps(KottiFieldInlineEdit.propsSchema),
+	// eslint-disable-next-line sonarjs/cognitive-complexity
+	setup(props, { emit }) {
+		onBeforeMount(() => {
+			window.addEventListener('click', onClick)
+			window.addEventListener('focus', onFocusChange, true)
+		})
+
+		onUnmounted(() => {
+			window.removeEventListener('click', onClick)
+			window.removeEventListener('focus', onFocusChange)
+		})
+
+		const { forceUpdate, forceUpdateKey } = useForceUpdate()
+		const translations = useTranslationNamespace('KtFieldInlineEdit')
+
+		const fieldRef = ref<_ComponentRef>(null)
+		const inputRef = ref<Nullable<FieldInlineEditElement>>(null)
+
+		const field = useField<KottiFieldInlineEdit.Value>({
+			emit,
+			isEmpty: (value) => value === null,
+			props,
+			supports: KOTTI_FIELD_INLINE_EDIT_SUPPORTS,
+		})
+
+		useAdjustHeight({
+			field,
+			inputRef,
+			props,
+		})
+
+		const fieldPlaceholder = computed(
+			() => props.placeholder ?? translations.value.placeholder,
+		)
+
+		const preventConfirm = computed<boolean>(() => {
+			const fieldValidation = field.validation.type
+
+			switch (props.preventConfirmationOn) {
+				case KottiFieldInlineEdit.ConfirmationValidation.ERROR:
+					return fieldValidation === 'error'
+				case KottiFieldInlineEdit.ConfirmationValidation.WARNING:
+					return ['error', 'warning'].includes(fieldValidation)
+				case KottiFieldInlineEdit.ConfirmationValidation.NEVER:
+					return false
+				default:
+					throw new Error(
+						`KtFieldInlineEdit: Unexpected type for 'preventConfirmationOn': ${props.preventConfirmationOn}`,
+					)
+			}
+		})
+
+		const setFieldValue = (newValue: KottiFieldInlineEdit.Value) => {
+			field.setValue(newValue === '' ? null : newValue)
+			forceUpdate()
+		}
+
+		const updateIsEditing = (
+			shouldEdit: KottiFieldInlineEdit.Events.UpdateIsEditing,
+		) => {
+			if (shouldEdit === props.isEditing) {
+				/*
+					condition shouldn't be moved outside of the wrapping if.
+					In case of not emitting the update:isEditing, the component
+					has to re-focus on the input field if needed.
+
+					Otherwise, the component emits and if the isEditing is set by
+					the user app, the same condition will be applied in the watcher.
+				*/
+				if (shouldEdit && !isInFocus(inputRef.value)) inputRef.value?.focus()
+
+				return
+			}
+
+			emit('update:isEditing', shouldEdit)
+		}
+
+		const preEditingValue = ref<KottiFieldInlineEdit.Value>(field.currentValue)
+
+		watch(
+			() => props.isEditing,
+			(shouldEdit, wasEditing) => {
+				if (shouldEdit && !isInFocus(inputRef.value)) inputRef.value?.focus()
+
+				if (shouldEdit === wasEditing) return
+
+				preEditingValue.value = field.currentValue
+			},
+			{ immediate: true },
+		)
+
+		watch(
+			() => field.currentValue,
+			(newValue) => {
+				if (!props.isEditing) preEditingValue.value = newValue
```

Comment:
Minor bug but we can ignore this for now. The app can’t update the value while the user is editing due to the chosen API, so if new data is loaded while the user is editing the cancel will actually reset it back to the old data

Reply:
no longer relevant i guess

Reply:
I don't imagine the app will allow the user to edit while the value meant is not loaded

we have isLoading to use when the app is aware of loading

Reply:
I found another issue though 

If the user app wants to update the value while the user is not editing, the BE won't be notified, because it likely only listens on confirm

so I'm considering emitting confirm in that case